dangerzone/gui/main_window.py

```diff
@@ -102,6 +156,119 @@ def __init__(self, dangerzone: DangerzoneGui) -> None:
 
         self.show()
 
+    def load_svg_image(self, filename: str) -> QtGui.QPixmap:
+        """Load an SVG image from a filename.
+
+        This answer is basically taken from: https://stackoverflow.com/a/25689790
+        """
+        path = get_resource_path(filename)
+        svg_renderer = QtSvg.QSvgRenderer(path)
+        image = QtGui.QImage(64, 64, QtGui.QImage.Format_ARGB32)
+        # Set the ARGB to 0 to prevent rendering artifacts
+        image.fill(0x00000000)
+        svg_renderer.render(QtGui.QPainter(image))
+        pixmap = QtGui.QPixmap.fromImage(image)
+        return pixmap
+
+    def show_update_success(self) -> None:
+        """Inform the user about a new Dangerzone release."""
+        version = self.dangerzone.settings.updater_get("latest_version")
+        changelog = self.dangerzone.settings.updater_get("latest_changelog")
+        update_widget = UpdateDialog(
+            self.dangerzone,
+            title=f"Dangerzone {version} has been released",
+            intro_msg=UPDATE_SUCCESS_MSG_INTRO,
+            html_msg=changelog,
+            epilogue_msg=UPDATE_SUCCESS_MSG_OUTRO,
+            ok_text="Ok",
+            has_cancel=False,
+        )
+        update_widget.exec_()
+
+    def show_update_error(self) -> None:
+        """Inform the user about an error during update checks"""
+        error = self.dangerzone.settings.updater_get("errors")[-1]
+        update_widget = UpdateDialog(
+            self.dangerzone,
+            title="Update check error",
+            intro_msg=UPDATE_ERROR_MSG_INTRO,
+            html_msg=error,
+            epilogue_msg=UPDATE_ERROR_MSG_OUTRO,
+            ok_text="Close",
+            has_cancel=False,
+        )
+        update_widget.exec_()
+
+    def toggle_updates_triggered(self) -> None:
+        """Change the underlying update check settings based on the user's choice."""
+        check = self.toggle_updates_action.isChecked()
+        self.dangerzone.settings.updater_set("check", check)
+        self.dangerzone.settings.save()
+
+    def handle_updates(self, report: dict) -> None:
+        """Handle update reports from the update checker thread.
+
+        See Updater.check_for_updates() to find the different types of reports that it
+        may send back, depending on the outcome of an update check.
+        """
+        # If there are no new updates, reset the error counter (if any) and return.
+        if not report:
+            if self.dangerzone.settings.updater_get("errors"):
+                self.dangerzone.settings.updater_set("errors", [])
+                self.dangerzone.settings.save()
+            return
+
+        hamburger_menu = self.hamburger_button.menu()
+
+        if report["error"]:
+            log.error(f"Encountered an error during an update check: {report['error']}")
+            # FIXME: Now that we have a notification bubble for update errors, does it
+            # make sense to store the errors in our settings?
```

Comment:
I'd store at most a counter. Not the messages themselves. When surfacing the error to the user, we could include the latest one found. It'll probably be similar to the past ones anyways.

Reply:
Fixed in 751441a.